saas/views/metrics.py

```diff
@@ -71,6 +74,47 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class CouponMetricsDownloadView(ProviderMixin, View):
+
+    headings = [
+        'Code',
+        'Percentage',
+        'Name',
+        'Email',
+        'Plan',
+    ]
+
+    def get(self, request, **kwargs):
+        class CouponAPIDummyView(SmartCouponListMixin, CouponAPIMixin):
+            '''
+            Stand-in for a view so that django-extra-views can create a
+            filtered QuerySet for us. This guarantees that we consider only the
+            same Coupons that are displayed to the user in the admin display.
+            '''
+            def __init__(self, request, kwargs):
+                self.request = request
+                self.kwargs = kwargs
+
+        coupons = CouponAPIDummyView(request, kwargs).get_queryset()
+
+        content = StringIO()
+        csv_writer = csv.writer(content)
+        csv_writer.writerow(self.headings)
+        for cartitem in CartItem.objects.filter(coupon__in=coupons):
+            csv_writer.writerow([
+                cartitem.coupon.code,
+                cartitem.coupon.percent,
+                ' '.join([cartitem.user.first_name, cartitem.user.last_name]),
```

Comment:
Good point. We should probably be UTF-8 encoding everything on the way out the door. I'll add that.